lib/storage/bucket.js

```diff
@@ -198,6 +198,10 @@ Bucket.prototype.delete = function(callback) {
  * var file = bucket.file('my-existing-file.png');
  */
 Bucket.prototype.file = function(name) {
+  if (!name) {
```

Comment:
The original diff as I remember it just checked if `bucket` param in File constructor was not false-y because @coderaiser was using the File constructor directly and not passing in a bucket parameter. Therefore they added the check for the bucket parameter. The current change now just moves the check of the filename (not the bucket like this PR title seems to suggest) from the File constructor into Bucket#file. This is actually worse I think because now File doesn't check its own params and Bucket#file is now responsible. The current change actually does nothing to help their original problem of not checking the bucket parameter.